numpy/testing/_private/utils.py

```diff
@@ -719,32 +738,19 @@ def chk_same_position(x_id, y_id, hasval='nan'):
             raise AssertionError(msg)
 
         if isnumber(x) and isnumber(y):
-            has_nan = has_inf = False
+            xy_isnan = xy_isinf = False
             if equal_nan:
-                x_isnan, y_isnan = isnan(x), isnan(y)
-                # Validate that NaNs are in the same place
-                has_nan = any(x_isnan) or any(y_isnan)
-                if has_nan:
-                    chk_same_position(x_isnan, y_isnan, hasval='nan')
+                xy_isnan = func_assert_same_pos(x, y, func=isnan, hasval='nan')
 
             if equal_inf:
-                x_isinf, y_isinf = isinf(x), isinf(y)
-                # Validate that infinite values are in the same place
-                has_inf = any(x_isinf) or any(y_isinf)
-                if has_inf:
-                    # Check +inf and -inf separately, since they are different
-                    chk_same_position(x == +inf, y == +inf, hasval='+inf')
-                    chk_same_position(x == -inf, y == -inf, hasval='-inf')
-
-            if has_nan and has_inf:
-                x = x[~(x_isnan | x_isinf)]
-                y = y[~(y_isnan | y_isinf)]
-            elif has_nan:
-                x = x[~x_isnan]
-                y = y[~y_isnan]
-            elif has_inf:
-                x = x[~x_isinf]
-                y = y[~y_isinf]
+                xy_isinf = func_assert_same_pos(x, y,
+                                                func=lambda xy: xy == +inf,
```

Comment:
`np.isposinf`?

Reply:
Learn something new every day...

Reply:
But doesn't work for complex numbers...

Reply:
Does what you have here work for complex numbers?